### PR TITLE
dotCMS/core#9287 UI: Create Login As functionality in NG2

### PR DESCRIPTION
**Pull request by fmontes**
**STILL NEED WORK TO CONNECT WITH THE APIS**

- UI work
- Create user service and load users into login as dot-select
- TSlint fixes

### CODE REVIEW

File: src/api/services/user-service.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import {ApiRoot} from "../persistence/ApiRoot";
+import {CoreWebService} from "./core-web-service";
+import {Injectable} from '@angular/core';
+import {Observable, Observer} from 'rxjs/Rx';
+import {RequestMethod, Http} from '@angular/http';
+
+@Injectable()
+export class UserService extends CoreWebService {
+    private userListUrl: string;
```
Comment (freddyucv) [inline src/api/services/user-service.ts]: The services should not extends from CoreWebService better inject it

File: src/view/boot.ts [hunk 2]
```diff
@@ -78,6 +77,7 @@ new AppConfigurationService().getConfigProperties().subscribe(config => {
         MdIconRegistry,
         provide('dotcmsConfig', {useValue: config.dotcmsConfig}),
         provide('routes', {useValue: routes}),
+        provide('user', {useValue: config.dotcmsConfig.configParams.user}),
```
Comment (freddyucv) [inline src/view/boot.ts]: Right now this user is storage in LoginService, we have to define how storage the current context.

File: src/view/components/common/login-as/login-as.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+@Component({
+    directives: [DotSelect, DotOption, MD_INPUT_DIRECTIVES, MdButton,],
+    moduleId: __moduleName,
+    providers: [UserService],
+    selector: 'dot-login-as',
+    styleUrls: ['login-as.css'],
+    templateUrl: ['login-as.html']
+})
+export class LoginAs {
```
Comment (freddyucv) [inline src/view/components/common/login-as/login-as.ts]: doesn't it should be LoginAsComponent?

File: src/view/components/common/toolbar-user/toolbar-user.ts [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+@Component({
+    directives: [DropdownComponent, LoginAs],
+    moduleId: __moduleName,
+    selector: 'toolbar-user',
+    styleUrls: ['toolbar-user.css'],
+    templateUrl: ['toolbar-user.html'],
+
+})
+export class ToolbarUser {
```
Comment (freddyucv) [inline src/view/components/common/toolbar-user/toolbar-user.ts]: doesn't it should be ToolbarComponent?

File: src/view/components/common/login-as/login-as.ts (hunk 3 above)
Comment (fmontes) [inline src/view/components/common/login-as/login-as.ts]: Yes!... my bad.

File: src/view/components/common/toolbar-user/toolbar-user.ts (hunk 4 above)
Comment (fmontes) [inline src/view/components/common/toolbar-user/toolbar-user.ts]: Yes.

File: src/view/components/common/toolbar-user/toolbar-user.html [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<dot-dropdown-component icon="account_circle" alignRight="true">
+    <ul class="user-menu">
+        <li class="user-menu__item">{{user.fullName}}</li>
+        <li class="user-menu__item"><a href="#" (click)="logout($event)">Logout</a></li>
+        <li class="user-menu__item"><a href="#" (click)="tooggleLoginAs($event)">Login as</a></li>
```
Comment (freddyucv) [inline src/view/components/common/toolbar-user/toolbar-user.html]: "Logout" and "Login as" label should be internationalized

Comment (fmontes) [inline src/view/components/common/toolbar-user/toolbar-user.html]: Yes... but we haven't define how are we doing this.

File: src/view/components/common/login-as/login-as.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+
+@Component({
+    directives: [DotSelect, DotOption, MD_INPUT_DIRECTIVES, MdButton,],
+    moduleId: __moduleName,
+    providers: [UserService],
+    selector: 'dot-login-as',
+    styleUrls: ['login-as.css'],
+    templateUrl: ['login-as.html']
+})
+export class LoginAsComponent {
```
Comment (fmontes) [inline src/view/components/common/login-as/login-as.ts]: Changed

File: src/view/components/common/toolbar-user/toolbar-user.ts [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+
+@Component({
+    directives: [DropdownComponent, LoginAsComponent],
+    moduleId: __moduleName,
+    selector: 'toolbar-user',
+    styleUrls: ['toolbar-user.css'],
+    templateUrl: ['toolbar-user.html'],
+
+})
+export class ToolbarUserComponent {
```
Comment (fmontes) [inline src/view/components/common/toolbar-user/toolbar-user.ts]: Changed

**Pull request closed without merging**